### PR TITLE
[IMP] mrp: allow disassembling through MO

### PR DESCRIPTION
**Pull request by rim-odoo**
- Task: https://www.odoo.com/web#id=5241&view_type=form&model=project.task&action=333&active_id=131
- Pad: https://pad.odoo.com/p/r.Vquhp9c9xsuPRWaX

### CODE REVIEW

**Comment by clonedagain** [comment]
We're interested in seeing this feature in Odoo, has there been progress lately?

File: addons/mrp/wizard/mrp_product_produce_view.xml [hunk 1]
```diff
@@ -9,8 +9,8 @@
             <field name="name">MRP Product Produce</field>
             <field name="model">mrp.product.produce</field>
             <field name="arch" type="xml">
-                <form string="Produce">
-                    <group string="Produce">
+                <form string="Produce" version="7.0">
```
Comment (hmo-odoo) [inline addons/mrp/wizard/mrp_product_produce_view.xml]: remove version=7.0

File: addons/mrp/wizard/mrp_product_produce.py [hunk 2]
```diff
@@ -73,6 +74,17 @@ def on_change_qty(self, cr, uid, ids, product_qty, consume_lines, context=None):
             new_consume_lines.append([0, False, consume])
         return {'value': {'consume_lines': new_consume_lines}}
 
+    def fields_view_get(self, cr, uid, view_id=None, view_type='form', context=None, toolbar=False, submenu=False):
```
Comment (hmo-odoo) [inline addons/mrp/wizard/mrp_product_produce.py]: really need fields_view_get method ? can not do with simple JS Widget ?

File: addons/mrp/wizard/mrp_product_produce.py [hunk 3]
```diff
@@ -21,7 +21,8 @@
 
 from openerp.osv import fields, osv
 import openerp.addons.decimal_precision as dp
-
+from lxml import etree
```
Comment (hmo-odoo) [inline addons/mrp/wizard/mrp_product_produce.py]: put in order

File: addons/mrp/wizard/disassemble_qty_view.xml [hunk 4]
```diff
@@ -0,0 +1,36 @@
+<?xml version="1.0" encoding="utf-8"?>
+<openerp>
```
Comment (hmo-odoo) [inline addons/mrp/wizard/disassemble_qty_view.xml]: replace with <odoo>

File: addons/mrp/wizard/disassemble_qty.py [hunk 5]
```diff
@@ -0,0 +1,63 @@
...
+#    You should have received a copy of the GNU Affero General Public License
+#    along with this program.  If not, see <http://www.gnu.org/licenses/>.
+#
+##############################################################################
+
+from openerp.osv import fields, osv
+from openerp.tools.translate import _
+import openerp.addons.decimal_precision as dp
+
+class change_disassemble_qty(osv.osv_memory):
```
Comment (hmo-odoo) [inline addons/mrp/wizard/disassemble_qty.py]: convert into new API standards

File: addons/mrp/test/disassemble_process.yml [hunk 6]
```diff
@@ -0,0 +1,96 @@
+-
+    MRP user can doing all process related to Production Order, so let's check data with giving the access rights of user.
```
Comment (hmo-odoo) [inline addons/mrp/test/disassemble_process.yml]: Migrate into python test case

File: addons/mrp/mrp.py [hunk 7]
```diff
@@ -109,6 +109,13 @@ def _check_capacity_per_cycle(self, cr, uid, ids, context=None):
         (_check_capacity_per_cycle, 'The capacity per cycle must be strictly positive.', ['capacity_per_cycle']),
     ]
 
+class stock_move(osv.osv):
```
Comment (hmo-odoo) [inline addons/mrp/mrp.py]: move this stuff into mrp/stock.py.

File: addons/mrp/wizard/mrp_product_produce_view.xml (hunk 1 above)
Comment (jpr-odoo) [inline addons/mrp/wizard/mrp_product_produce_view.xml]: Done!

File: addons/mrp/wizard/disassemble_qty_view.xml (hunk 4 above)
Comment (jpr-odoo) [inline addons/mrp/wizard/disassemble_qty_view.xml]: Done!

File: addons/mrp/wizard/disassemble_qty.py (hunk 5 above)
Comment (jpr-odoo) [inline addons/mrp/wizard/disassemble_qty.py]: Done! Migrated to the new api

File: addons/mrp/mrp.py (hunk 7 above)
Comment (jpr-odoo) [inline addons/mrp/mrp.py]: Done!

File: addons/mrp/wizard/mrp_product_produce.py (hunk 3 above)
Comment (jpr-odoo) [inline addons/mrp/wizard/mrp_product_produce.py]: Done!

File: addons/mrp/wizard/mrp_product_produce.py (hunk 2 above)
Comment (jpr-odoo) [inline addons/mrp/wizard/mrp_product_produce.py]: Done!

File: addons/mrp/test/disassemble_process.yml (hunk 6 above)
Comment (jpr-odoo) [inline addons/mrp/test/disassemble_process.yml]: Done!

**Comment by clonedagain** [comment]
@rim-odoo is this feature abandoned or has the development moved to another branch?

**Pull request closed without merging**